src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -470,21 +502,34 @@ class AssemblerConnectionManager(
             principal: Principal,
             columns: List<String>
     ): String {
-        val postgresUserName = if (principal.type == PrincipalType.USER) {
-            buildPostgresUsername(securePrincipalsManager.getPrincipal(principal.id))
-        } else {
-            buildPostgresRoleName(securePrincipalsManager.lookupRole(principal))
-        }
+        val postgresUserName =
+                if (principal.type == PrincipalType.USER) {
+                    buildPostgresUsername(securePrincipalsManager.getPrincipal(principal.id))
+                } else {
```

Comment:
just double checking -- `grantSelectSql` is only ever called on users and roles, right? To be safe / clear, you could make this an `} else if (principal.type === PrincipalType.ROLE) {` and add an `else { throw SomeException }`

Reply:
At this point, there shouldn't be any other type of principals, because we filter them in `Assembler.getReadAuthorizedUsersAndRolesOnEntitySet`, but yeah, you're right, I also it's a good idea to explicitly show that in the code too